pyccl/tests/test_power.py

```diff
@@ -8,6 +8,23 @@
 COSMO = ccl.Cosmology(
     Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
     transfer_function='bbks', matter_power_spectrum='halofit')
+COSMO_HM = ccl.Cosmology(
+    Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
+    transfer_function='bbks', matter_power_spectrum='halo_model',
+    mass_function='shethtormen')
+
+
+@pytest.mark.parametrize('k', [
+    1,
+    1.0,
+    [0.3, 0.5, 10],
+    np.array([0.3, 0.5, 10])
+])
+def test_nonlin_power_halomod(k):
+    a = 0.8
+    pk = ccl.nonlin_matter_power(COSMO_HM, k, a)
+    assert np.all(np.isfinite(pk))
+    assert np.shape(pk) == np.shape(k)
```

Comment:
Again it'd be good to test against a computation using the new classes in the test.

Reply:
Done